src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -28,22 +29,23 @@
      */
     public function rewind()
     {
-        $iterator = $this;
-        while ($iterator instanceof \OuterIterator) {
-            $innerIterator = $iterator->getInnerIterator();
+        if (PHP_VERSION_ID < 50523 || PHP_VERSION_ID >= 50600 && PHP_VERSION_ID < 50607) {
```

Comment:
You can use a guard close, it will improve the readability of this method:

```if (PHP_VERSION_ID > 50607 || (PHP_VERSION_ID > 50523 && PHP_VERSION_ID < 50600)) {
    parent:: rewind();
   return;
}
```